app/services/hearings_creator.rb

```diff
@@ -15,48 +15,49 @@ def call
 private
 
   def push_prosecution_cases
-    hearing[:prosecutionCases]&.each do |prosecution_case_data|
-      prosecution_case_data[:defendants].each do |defendant_data|
-        next if defendant_data[:offences].any? { |offence| offence.dig(:laaApplnReference, :applicationReference)&.start_with?("A", "Z") }
+    hearing_resulted.hearing.prosecution_cases.each do |prosecution_case|
+      prosecution_case.defendants.each do |defendant|
+        next if defendant.offences.any? do |offence|
+          LaaReference.new(maat_reference: offence.laa_reference_application_reference).dummy_maat_reference?
+        end
 
-        laa_reference = LaaReference.find_by!(defendant_id: defendant_data[:id], linked: true)
+        laa_reference = LaaReference.find_by(defendant_id: defendant.id, linked: true)
 
-        prosecution_case = MaatApi::ProsecutionCase.new(
-          hearing_body,
-          prosecution_case_data[:prosecutionCaseIdentifier][:caseURN],
-          defendant_data,
+        next if laa_reference.blank?
```

Comment:
Is this actually correct? I know we discussed it. But this refactor does actually change what we send to the queue.

Reply:
This makes sure we don't end up with failed jobs, but ideally we would not even check CDA's internal state at all.

Reply:
OK, right. I think I got confused thinking that those with blank `laa_references` would still get sent to MAATAPI. But they'll just hit a NoMethodError when we try to do `laa_reference.maat_reference` on line 32

Reply:
Not exactly. `next` will skip the rest of the code block and move onto the next item to iterate on (in this case, the next defendant in the list we're iterating over).

https://www.rubyguides.com/2019/09/ruby-next-break-keywords/

Reply:
Right, yes